src/fields/Fields.cpp

```diff
@@ -492,6 +499,58 @@ Fields::Copy (const int lev, const int i_slice, const amrex::Geometry& diag_geom
     }
 }
 
+void
+Fields::InitializeSlices (int lev, int islice, const amrex::Vector<amrex::Geometry>& geom)
+{
+    HIPACE_PROFILE("Fields::InitializeSlices()");
+
+    const bool explicit_solve = Hipace::GetInstance().m_explicit;
+    const bool deposit_rho = Hipace::GetInstance().m_deposit_rho;
+    const bool use_laser = Hipace::GetInstance().m_use_laser;
+
+    if (explicit_solve) {
+        if (lev != 0 && islice == geom[lev].Domain().bigEnd(Direction::z)) {
+            // first slice of lev (islice goes backwards)
+            // iterpolate jx_beam and jy_beam from lev-1 to lev
+            LevelUp(geom, lev, WhichSlice::Previous, "jx_beam");
+            LevelUp(geom, lev, WhichSlice::Previous, "jy_beam");
+            LevelUp(geom, lev, WhichSlice::This, "jx_beam");
+            LevelUp(geom, lev, WhichSlice::This, "jy_beam");
+            duplicate(lev, WhichSlice::This, {"jx"     , "jy"     },
+                           WhichSlice::This, {"jx_beam", "jy_beam"});
+        }
+        // Set all quantities to 0 except:
```

Comment:
```suggestion
        // Set all quantities on WhichSlice::This to 0 except:
```

This clarifies the comment below